<?xml version="1.0"?>
<launch>

  <env name="ROSCONSOLE_FORMAT" value="[${severity}][${node}][${function}:${line}]: ${message}"/>

  <arg name="robot_name"          default="woody"/>
  <arg name="x"                   default="0"/>
  <arg name="y"                   default="0"/>
  <arg name="yaw"                 default="0"/>
  <arg name="map_name"            default="regus"/>
  <arg name="simulation"          default="false"/>
  <arg name="gui_simulation"      default="false"/>
  <arg name="gui_teleop_keyboard" default="true"/>
  <arg name="gui_teleop_joy"      default="true"/>
  <arg name="gui_rviz"            default="true"/>
  <arg name="ekf"                 default="false"/>
  <arg name="rviz_file"           default="$(find robot_nav)/rviz/navigation.rviz"/>

  <node pkg="rosbag" type="record" name="rosbag" output="screen" 
  args="-j -q -o /ws_socialdroids/bag/$(arg robot_name) -e '/(amcl_pose|base_scan(.*)|clock|cmd_vel|joint_states|map|move_base(.*)|odom|robot_base(.*)|start_journay|tf|waypoints)'"/>

  <include file="$(find robot_description)/launch/bringup.launch" >
    <arg name="robot_name"          default="$(arg robot_name)"/>
    <arg name="x"                   default="$(arg x)"/>
    <arg name="y"                   default="$(arg y)"/>
    <arg name="yaw"                 default="$(arg yaw)"/>
    <arg name="world_name"          default="$(arg map_name)"/>
    <arg name="simulation"          default="$(arg simulation)"/>
    <arg name="gui_simulation"      default="$(arg gui_simulation)"/>
    <arg name="gui_teleop_keyboard" default="$(arg gui_teleop_keyboard)"/>
    <arg name="gui_teleop_joy"      default="$(arg gui_teleop_joy)"/>
    <arg name="gui_rviz"            default="$(arg gui_rviz)"/>
    <arg name="rviz_file"           default="$(arg rviz_file)"/>
  </include>

  <include file="$(find robot_nav)/launch/navigation.xml">
    <arg name="robot_name"          default="$(arg robot_name)"/>
    <arg name="y"                   default="$(arg x)"/>
    <arg name="x"                   default="$(arg y)"/>
    <arg name="yaw"                 default="$(arg yaw)"/>
    <arg name="map_name"            default="$(arg map_name)"/>

  </include>

  <node name="bumpers_control" pkg="robot_nav" type="bumpers.py" output="screen">
    <param name="time_duration" value="4.0"/>
    <param name="max_vel" value="1.0"/>
    <param name="odom_frame_id" value="/odom"/>
  </node>

  <node name="infrared_control" pkg="robot_nav" type="infrared.py" output="screen">
    <param name="time_duration" value="4.0"/>
    <param name="max_vel" value="1.0"/>
    <param name="odom_frame_id" value="/odom"/>
    <param name="threshold" value="0.2"/>
  </node>

<node name="follow_waypoints" pkg="robot_nav" type="follow.py" output="screen">
    <param name="init" value="1"/> <!--Init always (1) never (0)--> 
    <param name="goal_frame_id" value="map"/>
    <param name="wait_duration" value="0.0"/>
    <param name="xy_goal_tolerance" value="0.2"/>
    <param name="yaw_goal_tolerance" value="3.14"/>
    <param name="path_waypoints" value="$(arg map_name)"/>
    <param name="waypoint_distance_tolerance" value="0.0"/>
  </node>

  <group if="$(arg ekf)">
    <include file="$(find rf2o_laser_odometry)/launch/rf2o_laser_odometry.launch" >
    </include>

    <!-- Run the EKF Localization node -->
    <node pkg="robot_localization" type="ekf_localization_node" name="ekf_localization">
        <rosparam command="load" file="$(find robot_nav)/config/navigation/ekf_localization.yaml"/>
    </node>
  </group>

</launch>